server/src/main/java/org/opensearch/common/time/DateFormatter.java

```diff
@@ -126,6 +126,14 @@ default String formatJoda(DateTime dateTime) {
      */
     String pattern();
 
+    /**
+     * A name based format for this formatter. Can be one of the registered formatters like <code>epoch_millis</code> or
+     * a configured format like <code>HH:mm:ss</code>
+     *
+     * @return The name of this formatter
+     */
+    String printPattern();
```

Comment:
Why do we need this property?

Reply:
We need to decouple the datetime `format` for indexing and search. Right now the implicit assumption is that datetime fields will always be printed according to the first datetime format in default format if no format defined in mapping or the first one in the list of user defined formatters. I defined a `printFormat` that can be be defined while defing the formatter so that conversion of epoch back to datetime can be controlled.
For eg. while adding `strict_date_time_no_millis` to the default date time formatter as the first formatter was turning out to be a breaking change in terms of search because of this implicit assumption